Controller/CRUDController.php

```diff
@@ -313,8 +348,8 @@ public function deleteAction($id)
             } catch (ModelManagerException $e) {
                 $this->handleModelManagerException($e);
 
-                if ($this->isXmlHttpRequest()) {
-                    return $this->renderJson(array('result' => 'error'));
+                if ($this->isXmlHttpRequest($request)) {
+                    return $this->renderJson(array('result' => 'error'), 200, array(), $request);
```

Comment:
to me, it seems really strange to have `'result' => 'error'` and status code 200 in 1 response. However, it has always been 200.